pymc3/sampling.py

```diff
@@ -562,14 +593,27 @@ def sample(
     )
 
     if compute_convergence_checks:
+        # convert trace to InferenceData, with awareness of warmup!
+        # arviz 0.7.0 doesn't ignore warmup draws if they are in the trace! (see arviz #1146)
+        # that's why here, we make sure to only put actual posterior samples into idata.posterior
+        trace_warmup = trace[:-n_draws]   # <-- may result in len(trace_warmup) == 0
+        trace_posterior = trace[-n_draws:]
+        idata = arviz.from_pymc3(trace_posterior, log_likelihood=False, model=model)
```

Comment:
There should be an option to set `log_likelihood=True` in case the user wants to calculate loo/waic or whatever with the returned idata object. The hardcoded false made sense when the inference data generated was thrown away. Similar thing would go for tuning/warmup storage, but this should be solved once ArviZ PR warmup is merged

Reply:
Another option would be defaulting to None and set a `data.save_log_likelihood` as ArviZ rcParam for users who want to avoid memory overhead due to log_likelihood data.

Reply:
+1 for defaulting to None 👍 
People will indeed need the possibility of having the log likelihood in the idata (this is even the default on ArviZ, to compute LOO/WAIC).
I think the easiest way forward would be to hold this PR until ArviZ warmup PR is merged. That way, the different code blocks above can be properly refactored. WDYT?

Reply:
I've changed it such that it defaults to the ArviZ default or rcParams if set.

I'm resolving this thread - please comment on the latest changes.